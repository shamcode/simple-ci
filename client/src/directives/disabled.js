export default class {
    constructor() {
        this.node = null;
    }

    bind( node ) {
        this.node = node;
    }

    unbind() {
        this.node = null;
    }

    update( value ) {
        this.node.disabled = value;
    }
}
