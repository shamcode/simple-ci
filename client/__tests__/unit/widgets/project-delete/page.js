import { DI } from 'sham-ui';
import Page from '../../../../src/widgets/project-delete/page';
import disabled from '../../../../src/directives/disabled';
import directives from '../../../../src/directives/event-listener';
import renderer from 'sham-ui-test-helpers';

const flushPromises = () => new Promise( resolve => setImmediate( resolve ) );

afterEach( () => {
    DI.bind( 'store', null );
    DI.bind( 'router', null );
} );

it( 'renders correctly', () => {
    const storeMock = jest.fn().mockReturnValue( Promise.resolve( { project: { id: 1 } } ) );
    DI.bind( 'store', {
        getProjectById: storeMock
    } );
    DI.bind( 'router', {
        lastRouteResolved: jest.fn().mockReturnValueOnce( { params: { id: 1 } } ),
        generate: jest.fn().mockReturnValueOnce( '/' )
    } );

    const meta = renderer( Page, {
        directives: {
            disabled,
            ...directives
        }
    } );
    expect( storeMock.mock.calls[ 0 ][ 0 ] ).toBe( 1 );
    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'render errors', async () => {
    expect.assertions( 1 );

    const storeMock = jest.fn().mockReturnValue( Promise.reject() );
    DI.bind( 'store', {
        getProjectById: storeMock
    } );
    DI.bind( 'router', {
        lastRouteResolved: jest.fn().mockReturnValueOnce(),
        generate: jest.fn().mockReturnValueOnce( '/' )
    } );

    const meta = renderer( Page, {
        directives: {
            disabled,
            ...directives
        }
    } );
    await flushPromises();

    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'delete fail', async() => {
    expect.assertions( 4 );

    const deleteProject = jest.fn();
    DI.bind( 'store', {
        getProjectById: jest.fn().mockReturnValue( Promise.resolve( {
            project: {
                id: 1,
                name: 'Test',
                cwd: '/tmp/'
            }
        } ) ),
        deleteProject: deleteProject
    } );
    const navigateMock = jest.fn();
    DI.bind( 'router', {
        lastRouteResolved: jest.fn().mockReturnValue( { params: { id: 1 } } ),
        generate: jest.fn().mockReturnValue( '/' ),
        navigate: navigateMock
    } );

    const meta = renderer( Page, {
        directives: {
            disabled,
            ...directives
        }
    } );
    const { widget: { container } } = meta;
    await flushPromises();

    deleteProject.mockReturnValueOnce( Promise.reject() );
    container.querySelector( '.ok' ).click();
    await flushPromises();

    expect( deleteProject.mock.calls.length ).toBe( 1 );
    expect( navigateMock.mock.calls.length ).toBe( 1 );
    expect( navigateMock.mock.calls[ 0 ][ 0 ] ).toBe( '/' );
    expect( meta.toJSON() ).toMatchSnapshot();
} );

it( 'cancel', async() => {
    expect.assertions( 2 );

    DI.bind( 'store', {
        getProjectById: jest.fn().mockReturnValue( Promise.resolve( {
            project: {
                id: 1,
                name: 'Test',
                cwd: '/tmp/'
            }
        } ) )
    } );
    const navigateMock = jest.fn();
    DI.bind( 'router', {
        lastRouteResolved: jest.fn().mockReturnValue( { params: { id: 1 } } ),
        generate: jest.fn().mockReturnValue( '/' ),
        navigate: navigateMock
    } );

    const meta = renderer( Page, {
        directives: {
            disabled,
            ...directives
        }
    } );
    const { widget: { container } } = meta;
    await flushPromises();

    container.querySelector( '.cancel' ).click();
    expect( navigateMock.mock.calls.length ).toBe( 1 );
    expect( navigateMock.mock.calls[ 0 ][ 0 ] ).toBe( '/' );

    await flushPromises();
} );