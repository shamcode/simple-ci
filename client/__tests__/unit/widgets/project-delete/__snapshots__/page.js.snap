// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`delete fail 1`] = `
Object {
  "ID": "widget",
  "constructorOptions": Object {
    "directives": Object {
      "disabled": [Function],
      "onclick": [Function],
      "onsubmit": [Function],
    },
  },
  "html": "
<!--if-->
<!--ErrorsPanel-->
<div class=\\"modal\\">
  <div class=\\"content\\">
    <div class=\\"header\\"> Delete project </div>
    <div class=\\"body\\"> Do you want delete project \\"Test\\"? </div>
    <div class=\\"footer\\">
      <button class=\\"ok danger\\" disabled=\\"\\"><span>Confirm</span></button>
      <button class=\\"cancel\\"><span>Cancel</span></button>
    </div>
  </div>
</div>
<!--Modal-->
<!--if-->
<!--if-->
",
  "nodes": Array [
    Object {
      "ID": undefined,
      "constructorOptions": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
  ],
  "options": Object {
    "actionSequence": Array [
      "render",
      "bindEvents",
    ],
    "afterRender": [Function],
    "dataLoaded": true,
    "dataSaving": true,
    "directives": Object {
      "disabled": [Function],
      "onclick": [Function],
      "onsubmit": [Function],
    },
    "errors": Array [],
    "project": Object {
      "cwd": "/tmp/",
      "id": 1,
      "name": "Test",
    },
    "renderAsync": false,
    "renderDependence": Array [],
    "types": Array [],
  },
}
`;

exports[`render errors 1`] = `
Object {
  "ID": "widget",
  "constructorOptions": Object {
    "directives": Object {
      "disabled": [Function],
      "onclick": [Function],
      "onsubmit": [Function],
    },
  },
  "html": "
<ul class=\\"errors-panel\\">
  <li>Load project fail!</li>
</ul>
<!--if-->
<!--ErrorsPanel-->
<!--if-->
<!--if-->
",
  "nodes": Array [
    Object {
      "ID": undefined,
      "constructorOptions": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
  ],
  "options": Object {
    "actionSequence": Array [
      "render",
      "bindEvents",
    ],
    "afterRender": [Function],
    "dataLoaded": true,
    "dataSaving": false,
    "directives": Object {
      "disabled": [Function],
      "onclick": [Function],
      "onsubmit": [Function],
    },
    "errors": Array [
      "Load project fail!",
    ],
    "project": Object {},
    "renderAsync": false,
    "renderDependence": Array [],
    "types": Array [],
  },
}
`;

exports[`renders correctly 1`] = `
Object {
  "ID": "widget",
  "constructorOptions": Object {
    "directives": Object {
      "disabled": [Function],
      "onclick": [Function],
      "onsubmit": [Function],
    },
  },
  "html": "
<!--if-->
<!--FlushPageDataLoading-->
<!--if-->
",
  "nodes": Array [
    Object {
      "ID": undefined,
      "constructorOptions": null,
      "html": null,
      "nodes": null,
      "options": null,
    },
  ],
  "options": Object {
    "actionSequence": Array [
      "render",
      "bindEvents",
    ],
    "afterRender": [Function],
    "dataLoaded": false,
    "dataSaving": false,
    "directives": Object {
      "disabled": [Function],
      "onclick": [Function],
      "onsubmit": [Function],
    },
    "errors": Array [],
    "project": Object {},
    "renderAsync": false,
    "renderDependence": Array [],
    "types": Array [],
  },
}
`;
